utils/retryexecutor.go

```diff
@@ -39,10 +37,15 @@ func (runner *RetryExecutor) Execute() error {
 			return err
 		}
 
-		log.Warn(runner.getLogRetryMessage(i, err))
+		message := runner.getLogRetryMessage(i+1, err)
```

Comment:
The logging logic is currently splinted between this place and the ```getLogRetryMessage``` function. I suggest mering everything into one function, that will do everything, named ```logRetry```. This shouls make the code reading easier.